autorelay: flaky TestMaxAge

https://github.com/libp2p/go-libp2p/runs/7748138576?check_suite_focus=true

```
  === RUN   TestMaxAge
      autorelay_test.go:404: 
          	Error Trace:	/home/runner/work/go-libp2p/go-libp2p/p2p/host/autorelay/autorelay_test.go:404
          	Error:      	Condition never satisfied
          	Test:       	TestMaxAge
  --- FAIL: TestMaxAge (6.52s)

```
=== RUN   TestMaxAge
2022-08-10T09:40:48.841+0200	DEBUG	autorelay	autorelay/relay_finder.go:653	starting relay finder
2022-08-10T09:40:48.841+0200	DEBUG	autorelay	autorelay/relay_finder.go:246	found node	{"id": "12D3KooWAGEPewgrHmWnvmbt1GArsWGBXBiQo1mEu23tb2iJdZRd"}
2022-08-10T09:40:48.841+0200	DEBUG	autorelay	autorelay/relay_finder.go:246	found node	{"id": "12D3KooWSjUQeHBQBrkrxFDJR7cHShHiEtUwRyvq2dfegDar2UH4"}
2022-08-10T09:40:48.841+0200	DEBUG	autorelay	autorelay/relay_finder.go:246	found node	{"id": "12D3KooWFvvevPfYzG6iQ63t7s7cYkxTjzBYGZvFhgQUqkt4Lxfu"}
2022-08-10T09:40:48.841+0200	DEBUG	autorelay	autorelay/relay_finder.go:246	found node	{"id": "12D3KooWFuhZDV9yucDace2RPKfqy7nZ8JoMpGpBjqxoRMF5VkLf"}
2022-08-10T09:40:48.847+0200	DEBUG	autorelay	autorelay/relay_finder.go:336	node supports relay protocol	{"peer": "12D3KooWFuhZDV9yucDace2RPKfqy7nZ8JoMpGpBjqxoRMF5VkLf", "supports circuit v2": true}
2022-08-10T09:40:48.848+0200	DEBUG	autorelay	autorelay/relay_finder.go:484	adding new relay	{"id": "12D3KooWFuhZDV9yucDace2RPKfqy7nZ8JoMpGpBjqxoRMF5VkLf"}
2022-08-10T09:40:48.848+0200	DEBUG	autorelay	autorelay/relay_finder.go:336	node supports relay protocol	{"peer": "12D3KooWSjUQeHBQBrkrxFDJR7cHShHiEtUwRyvq2dfegDar2UH4", "supports circuit v2": true}
2022-08-10T09:40:48.849+0200	DEBUG	autorelay	autorelay/relay_finder.go:336	node supports relay protocol	{"peer": "12D3KooWAGEPewgrHmWnvmbt1GArsWGBXBiQo1mEu23tb2iJdZRd", "supports circuit v2": true}
2022-08-10T09:40:48.849+0200	DEBUG	autorelay	autorelay/relay_finder.go:336	node supports relay protocol	{"peer": "12D3KooWFvvevPfYzG6iQ63t7s7cYkxTjzBYGZvFhgQUqkt4Lxfu", "supports circuit v2": true}
2022-08-10T09:40:48.849+0200	DEBUG	autorelay	autorelay/relay_finder.go:224	not enough candidates. Resetting timer	{"num": 3, "desired": 4}
2022-08-10T09:40:48.849+0200	DEBUG	autorelay	autorelay/relay_finder.go:224	not enough candidates. Resetting timer	{"num": 3, "desired": 4}
2022-08-10T09:40:49.073+0200	DEBUG	autorelay	autorelay/relay_finder.go:246	found node	{"id": "12D3KooW9qsLYyDksc44nmTbx8hvFbnRaQKCJ3MprPpvxXjh5KAG"}
2022-08-10T09:40:49.073+0200	DEBUG	autorelay	autorelay/relay_finder.go:246	found node	{"id": "12D3KooWNFNpEoYj54LJwF2Qfbz8bb1hR9zqMS6AJuShqp8vxEBr"}
2022-08-10T09:40:49.073+0200	DEBUG	autorelay	autorelay/relay_finder.go:246	found node	{"id": "12D3KooWEBym6mfkpvBxtoGY2EyvarEtUz9A7Y2GWfQnVvuxef5T"}
2022-08-10T09:40:49.074+0200	DEBUG	autorelay	autorelay/relay_finder.go:246	found node	{"id": "12D3KooWDMTygVfb3qkVEVQXGxuRS5NyhEdLWqCEGWsQutJDndcJ"}
2022-08-10T09:40:49.087+0200	DEBUG	autorelay	autorelay/relay_finder.go:336	node supports relay protocol	{"peer": "12D3KooW9qsLYyDksc44nmTbx8hvFbnRaQKCJ3MprPpvxXjh5KAG", "supports circuit v2": true}
2022-08-10T09:40:49.087+0200	DEBUG	autorelay	autorelay/relay_finder.go:336	node supports relay protocol	{"peer": "12D3KooWEBym6mfkpvBxtoGY2EyvarEtUz9A7Y2GWfQnVvuxef5T", "supports circuit v2": true}
2022-08-10T09:40:49.087+0200	DEBUG	autorelay	autorelay/relay_finder.go:336	node supports relay protocol	{"peer": "12D3KooWNFNpEoYj54LJwF2Qfbz8bb1hR9zqMS6AJuShqp8vxEBr", "supports circuit v2": true}
2022-08-10T09:40:49.088+0200	DEBUG	autorelay	autorelay/relay_finder.go:336	node supports relay protocol	{"peer": "12D3KooWDMTygVfb3qkVEVQXGxuRS5NyhEdLWqCEGWsQutJDndcJ", "supports circuit v2": true}
2022-08-10T09:40:49.103+0200	DEBUG	autorelay	autorelay/relay_finder.go:184	deleting candidate due to age	{"id": "12D3KooWSjUQeHBQBrkrxFDJR7cHShHiEtUwRyvq2dfegDar2UH4"}
2022-08-10T09:40:49.103+0200	DEBUG	autorelay	autorelay/relay_finder.go:184	deleting candidate due to age	{"id": "12D3KooWAGEPewgrHmWnvmbt1GArsWGBXBiQo1mEu23tb2iJdZRd"}
2022-08-10T09:40:49.103+0200	DEBUG	autorelay	autorelay/relay_finder.go:184	deleting candidate due to age	{"id": "12D3KooWFvvevPfYzG6iQ63t7s7cYkxTjzBYGZvFhgQUqkt4Lxfu"}
    autorelay_test.go:421: 
        	Error Trace:	/home/parallels/src/go/src/github.com/libp2p/go-libp2p/p2p/host/autorelay/autorelay_test.go:421
        	Error:      	Condition never satisfied
        	Test:       	TestMaxAge
2022-08-10T09:40:52.121+0200	DEBUG	autorelay	autorelay/relay_finder.go:667	stopping relay finder
--- FAIL: TestMaxAge (3.39s)

--- a/p2p/host/autorelay/autorelay_test.go
+++ b/p2p/host/autorelay/autorelay_test.go
@@ -400,8 +400,11 @@ func TestMaxAge(t *testing.T) {
 	require.Eventually(t, func() bool { return numRelays(h) > 0 }, 3*time.Second, 100*time.Millisecond)
 	relays := usedRelays(h)
 	require.Len(t, relays, 1)
-	cl.Add(time.Second)
-	require.Eventually(t, func() bool { return len(peerChans) == 0 }, time.Second, 100*time.Millisecond)
+	require.Eventually(t, func() bool {
+		// we don't know exactly when the timer is reset, just advance our timer multiple times if necessary
+		cl.Add(time.Second)
+		return len(peerChans) == 0
+	}, 500*time.Millisecond, 100*time.Millisecond)
 
 	cl.Add(10 * time.Minute)
 	for _, r := range relays2 {

